modules/monitor/software/channel_monitor.cc

```diff
@@ -53,79 +54,54 @@ namespace apollo {
 namespace monitor {
 namespace {
 
+using ReaderAndMessagePair =
+    std::pair<std::shared_ptr<cyber::ReaderBase>,
+              std::shared_ptr<google::protobuf::Message>>;
+
+template <typename T>
+ReaderAndMessagePair CreateReaderAndLatestsMessage(const std::string& channel) {
+  const auto reader = MonitorManager::Instance()->CreateReader<T>(channel);
```

Comment:
Type of reader should be `auto`  rather than `const auto`. 
See
https://github.com/ApolloAuto/apollo/blob/master/cyber/node/reader_base.h#L73

Reply:
The reader is a constant shared pointer object. Getting the internal pointer from shared pointer object is a "const" operation. Otherwise the compiler would have complained. The previous version of code use the same const auto reader.

Reply:
OK, I see. `const` was for shared_ptr rather than ReaderBase class instance.